distmv/general.go

```diff
@@ -0,0 +1,21 @@
+package distmv
+
+var (
+	ErrSizeMismatch  = "dist: size mismatch"
+	ErrZeroDimension = "dist: zero dimensional input"
+	ErrReceiver      = "dist: input slice is not nil or the correct length"
+)
+
+const logTwoPi = 1.8378770664093454835606594728112352797227949472755668
+
+// useAs gets a slice of size n. If len(x) == n, x is returned, if x == nil
+// a new slice is allocated, otherwise useAs panics
+func useAs(x []float64, n int) []float64 {
+	if len(x) == n {
```

Comment:
Do you really want this exact? In mat64 the test is `n < cap(f)` and then it's resliced to n.